spark/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -562,6 +562,203 @@ class DeltaTable private[tables](
+   *   true,
+   *   Map("foo" -> "bar"))
+   * }}}
+   *
+   * @param target The path or table name to create the clone.
+   * @param replace Whether to replace the destination with the clone command.
+   * @param properties The table properties to override in the clone.
+   * @since 3.3.0
+   */
+  def clone(target: String, replace: Boolean, properties: Map[String, String]): DeltaTable = {
+    executeClone(table, target, replace, properties)
+  }
+
+  /**
+   * Clone a DeltaTable to a given destination to mirror the existing table's data and metadata.
+   * 
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.clone("/some/path", true)
+   * }}}
+   *
+   * @param target The path or table name to create the clone.
+   * @param replace Whether to replace the destination with the clone command.
+   * @since 3.3.0
+   */
+  def clone(target: String, replace: Boolean): DeltaTable = {
+    clone(target, replace, properties = Map.empty)
+  }
+
+  /**
+   * Clone a DeltaTable to a given destination to mirror the existing table's data and metadata.
+   * 
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.clone("/some/path")
+   * }}}
+   *
+   * @param target The path or table name to create the clone.
+   *
+   * @since 3.3.0
+   */
+  def clone(target: String): DeltaTable = {
+    clone(target, replace = false)
+  }
+
+  /**
+   * Clone a DeltaTable at a specific version to a given destination to mirror the existing
+   * table's data and metadata.
+   *
+   * Specifying properties here means that the target will override any properties with the same key
+   * in the source table with the user-defined properties.
+   *
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.cloneAtVersion(
+   *   5,
+   *   "/some/path",
+   *   true,
+   *   Map("foo" -> "bar"))
+   * }}}
+   *
+   * @param version The version of this table to clone from.
+   * @param target The path or table name to create the clone.
+   * @param replace Whether to replace the destination with the clone command.
+   * @param properties The table properties to override in the clone.
+   *
+   * @since 3.3.0
+   */
+  def cloneAtVersion(
+      version: Long,
+      target: String,
+      replace: Boolean,
+      properties: Map[String, String]): DeltaTable = {
+    executeClone(table, target, replace, properties, versionAsOf = Some(version))
+  }
+
+  /**
+   * Clone a DeltaTable at a specific version to a given destination to mirror the existing
+   * table's data and metadata.
+   * 
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.cloneAtVersion(5, "/some/path", true)
+   * }}}
+   *
+   * @param version The version of this table to clone from.
+   * @param target The path or table name to create the clone.
+   * @param replace Whether to replace the destination with the clone command.
+   *
+   * @since 3.3.0
+   */
+  def cloneAtVersion(version: Long, target: String, replace: Boolean): DeltaTable = {
```

Comment:
Could we add a counterpart `Int` `version` argument for all `cloneAtVersion` functions?

```suggestion

cloneAtVersion(version: Long, target: String, replace: Boolean, properties: Map[String, String])

cloneAtVersion(version: Int, target: String, replace: Boolean, properties: Map[String, String]) = {
   cloneAtVersion(version.toLong, target, replace, properties)
}

cloneAtVersion(version: Long, target: String, replace: Boolean)
cloneAtVersion(version: Int, target: String, replace: Boolean)
cloneAtVersion(version: Long, target: String, replace: Boolean)
cloneAtVersion(version: Int, target: String, replace: Boolean)

```

And so on

Reply:
The only precedent is `restoreToVersion` which only takes a Long as the version, that seems like a lot of bloated extra APIs

Reply:
@Kimahriman I think having many APIs as fine as long as it makes things easier for the users right? I don't think from the user perspective, having 5 or 10 APIs don't really affect their experinence.

Reply:
@Kimahriman For `merge` and `delete` for examples, we accept both `String` and `Column`

Reply:
Based on quick testing both Scala and Java will implicitly convert an int to a long for you, so there's no need for an additional API

Reply:
@Kimahriman Okay, we are good then